pkg/pillar/types/dns.go

```diff
@@ -682,33 +685,40 @@ func getLocalAddrListImpl(dns DeviceNetworkStatus,
 }
 
 // Check if an interface name is a port owned by nim
-func IsPort(dns DeviceNetworkStatus, ifname string) bool {
+func IsPort(dns DeviceNetworkStatus, ifname string, usbaddr string, usbprod string, pciaddr string) bool {
 	for _, us := range dns.Ports {
 		if us.IfName != ifname {
```

Comment:
@andrewd-zededa As additional fix, we might need to apply this check only if ifname is non-empty, i.e.
```
if ifname != "" && us.IfName != ifname {
  continue
}
```
Even when wwan adapter has empty ifname in the model (because PCI or USB address is used instead to match it), DPCManager will still report the interface name in DeviceNetworkStatus once it is [discovered](https://github.com/lf-edge/eve/blob/master/pkg/pillar/dpcmanager/wwan.go#L134). (we need interface name for metrics etc.)
So to avoid returning false for wwan adapter with empty ifname in the model but actually used by NIM, match by ifname only if non empty.

Otherwise the PR LGTM.

Reply:
@milan-zededa ok modified all the ifname compares to also check if not empty, in the latest push.